sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusReceivedMessage.cs

```diff
@@ -14,21 +16,58 @@ namespace Azure.Messaging.ServiceBus
     /// </summary>
     public class ServiceBusReceivedMessage
     {
+
+        /// <summary>
+        /// Creates a new message from the specified payload.
+        /// </summary>
+        /// <param name="body">The payload of the message in bytes</param>
+        internal ServiceBusReceivedMessage(ReadOnlyMemory<byte> body)
+        {
+            AmqpMessage = new AmqpAnnotatedMessage(new BinaryData[] { BinaryData.FromBytes(body) });
```

Comment:
nit: use `this(new AmqpAnnotatedMessage(new BinaryData[] { BinaryData.FromBytes(body) }))` since you have a constructor that takes AmqpAnnotatedMessage.